src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -601,6 +609,40 @@ public static DType AttachDataSourceInfo(DType type, IExternalTabularDataSource
                 returnType = returnType.SetType(ref fError, DPath.Root.Append(typedName.Name), AttachDataSourceInfo(typedName.Type, dsInfo, false), skipCompare: true);
             }
 
+            return returnType;
+        }              
+
+        /// <summary>
+        /// This should only be used when constructing DTypes from the public surface to replace an existing display name provider
+        /// </summary>
+        public static DType ReplaceDisplayNameProvider(DType type, DisplayNameProvider displayNames)
+        {
+            type.AssertValid();
+            Contracts.AssertValue(displayNames);
+
+            DType returnType = type.Clone();
+            returnType.DisplayNameProvider = displayNames;
+            return returnType;
+        }
+               
+
+        /// <summary>
+        /// This should be used by internal operations to update the set of display name providers associated with a type, i.e. during Union operations.
+        /// </summary>
+        public static DType AttachDisplayNameProvider(DType type, DisplayNameProvider displayNames)
```

Comment:
This function actually disables display name providers in some cases, but this is not implied by the function name.

Reply:
updated name and doc comment